toxics/limit_data.go

```diff
@@ -0,0 +1,60 @@
+package toxics
+
+import "github.com/Shopify/toxiproxy/stream"
+
+// LimitDataToxic has limit in bytes
+type LimitDataToxic struct {
+	Bytes int64 `json:"bytes"`
+}
+
+type LimitDataToxicState struct {
+	bytesTransmitted int64
+}
+
+func (t *LimitDataToxic) Pipe(stub *ToxicStub) {
+	state := stub.State.(*LimitDataToxicState)
+	var bytesRemaining = t.Bytes - state.bytesTransmitted
```

Comment:
Toxics can be interrupted at any point and restarted, so some bytes may have been transmitted, the toxic is updated, and then `Pipe()` is restarted, but with the same state as before.